File: src/main/java/com/szadowsz/formstar/star/Star.java
package com.szadowsz.formstar.star;

import com.szadowsz.formstar.calc.StarCalc;

import java.util.Random;

/**
 * Class to Represent a Star
 * 
 * @author zakski
 *
 */
public class Star {

	public double stellarMass; // weight of the star in Solar Mass
	public double stellarLuminosity; // brightness of the star in Solar Luminosity
	public double stellarRadius; // radius of the star in Solar Radius
	public double stellarTemp; // temperature of the star in kelvin
	public double mainSeqLife; // Major Sequence Lifespan in terms of years
	public double age; // age of the star
	public double radiusEcosphere; // outer limit of the Star's Habitable zone
	public double radiusGreenhouse; // inner limit of the Star's Habitable zone
	String stellarClass; // the class of the sun. TODO Get Methods to calculate this / create startype enums

	/**
	 * Method to procedurally generate a star
	 * 
	 * @param rand - pseudo-random number generator interface
	 * @return - the generated star
	 */
	public static Star generate(Random rand){
		return StarCalc.formStar(rand);
	}
	
}

File: src/main/scala/com/szadowsz/accrete/fogg/bodies/Star.scala
package com.szadowsz.accrete.fogg.bodies

import com.szadowsz.accrete.fogg.calc.FoggAccrCalc

import scala.util.Random

/**
 * @author Zakski : 06/07/2015.
 */
class Star(calc: FoggAccrCalc, rand: Random) {

  val mass: Double = calc.stellarMass(rand)

  val luminosity: Double = calc.stellarLuminosity(mass)
  val age: Double = calc.stellarAge(rand, _lifespan)
  val meanHabitableRadius: Double = calc.ecosphereRadius(luminosity)
  val innerHabitableRadius: Double = calc.greenhouseRadius(meanHabitableRadius)
  private val _lifespan: Double = calc.stellarMSLifespan(mass, luminosity)
}
